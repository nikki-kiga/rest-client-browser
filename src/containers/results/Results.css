.results {
    background: #f5f5f5;
    border: 1px solid #111;
    flex: 1;
    height: calc(100vh - 18rem);
    margin: 1rem;
    overflow: auto;
    padding: 2rem;
  }
