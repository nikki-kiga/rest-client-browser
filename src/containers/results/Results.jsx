import React from 'react';
import PropTypes from 'prop-types';
import styles from './Results.css';

const Result = ({ response }) => {
  return (
    <section className={styles.results}>
      <h2>Results</h2>
      {
        Object.entries(response).length === 0 ? null : <p>{JSON.stringify(response, null, 2)}</p>
      }    
    </section>
  );
};

Result.propTypes = {
  response: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
};

export default Result;
