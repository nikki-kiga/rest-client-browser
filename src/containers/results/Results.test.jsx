import React from 'react';
import { shallow } from 'enzyme';
import Results from './Results.jsx';

describe('Results container', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Results response={{}} />);
  });

  it('renders a results container', () => {
    expect(wrapper).toMatchSnapshot();
  });

});
