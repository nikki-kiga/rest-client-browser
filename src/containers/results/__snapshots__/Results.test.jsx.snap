// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results container renders a results container 1`] = `
<section
  className="results"
>
  <h2>
    Results
  </h2>
</section>
`;
